### DISCUSSION TITLE
How to search on a TV with just a remote.

### DISCUSSION THREAD

**Discussion by jean-louis67** [photoprism/photoprism#3557] [Q&A]
Hello

I try to use PhotoPrism via the browser of a TV connected to WIFI.
It works well. 
When necessary, a keyboard is displayed on which there is no return key.
When I do a search, I enter some text but can't start the search, since there is no button for that in PhotoPrism.

Thanks in advance for your suggestions!
Jean-Louis

**Comment by lastzero** [comment]
Unfortunately, we didn't have much time to optimize the user interface for use on a smart TV. When I tested it about a year ago, I noticed that certain browsers worked much better than others.

From what I remember, I tested Firefox and the Silk browser installed on our Amazon Fire TV Stick, although I'm unsure which one worked better.

**Reply by lastzero** [reply]
That said, it would of course be possible to add a search button or otherwise optimize the UI for use on a TV.

**Comment by jean-louis67** [comment]
Thank you very much for your quick response!
In my case it is a Samsung TV and its own browser. I don't think it's the browser itself that's to blame, but rather the virtual keyboard (or maybe PhotoPrism :wink:). Maybe something escaped me. There is a Goto button that appears with the keyboard. When this is used to enter a URL in the address bar, the keyboard disappears and the browser will display the requested URL. On the other hand, in the case of a search with PhotoPrism, the keyboard also disappears, but the search is not carried out.
I don't know if PhotoPrism directly receives a sequence of characters (for example LF and/or CR LF) or an event.

It is true that an explicit button would solve the problem. It is sometimes considered that the use of a keyboard key to trigger an action must necessarily be the shortcut of a menu or a clearly accessible button. 
There is also the possibility of a dynamic search as the string is entered, but I know this is debatable.

That said, the use of the virtual keyboard is very tedious, perhaps a mini WIFI keyboard could be useful and that's in my backyard!

**Comment by jean-louis67** [comment]
Hello,
My problem is solved: I no longer use the virtual keyboard offered by the browser. Instead I can choose to use the keyboard of my smart phone. And in this configuration, searches are carried out correctly.
I don't know if being able to use my smart phone in this case is general. In my case Smart phone and Smart TV are from the same manufacturer.
